targets/csharp/make.js

```diff
@@ -262,36 +262,49 @@ function getAuthParams(apiCall, isApiInstance = false) {
+        return tabbing + "request.TitleId = request.TitleId ?? PlayFabSettings.TitleId;\n"
             + tabbing + "if (request.TitleId == null) throw new PlayFabException(PlayFabExceptionCode.TitleNotSet, \"Must be have PlayFabSettings.TitleId set to call this method\");\n";
     if (apiCall.auth === "EntityToken")
-        return tabbing + "if (PlayFabSettings.EntityToken == null) throw new PlayFabException(PlayFabExceptionCode.EntityTokenNotSet, \"Must call GetEntityToken before calling this method\");\n";
+        return tabbing + "if ((request.AuthenticationContext?.EntityToken ?? PlayFabSettings.EntityToken) == null) throw new PlayFabException(PlayFabExceptionCode.EntityTokenNotSet, \"Must call GetEntityToken before calling this method\");\n";
     if (apiCall.auth === "SessionTicket")
-        return tabbing + "if (PlayFabSettings.ClientSessionTicket == null) throw new PlayFabException(PlayFabExceptionCode.NotLoggedIn, \"Must be logged in to call this method\");\n";
+        return tabbing + "var clientSessionTicket = request.AuthenticationContext?.ClientSessionTicket ?? PlayFabSettings.ClientSessionTicket;\n            if (clientSessionTicket == null) throw new PlayFabException(PlayFabExceptionCode.NotLoggedIn, \"Must be logged in to call this method\");\n";
     if (apiCall.auth === "SecretKey" && isApiInstance === false)
         return tabbing + "if (PlayFabSettings.DeveloperSecretKey == null) throw new PlayFabException(PlayFabExceptionCode.DeveloperKeyNotSet, \"Must have PlayFabSettings.DeveloperSecretKey set to call this method\");\n";
     if (apiCall.auth === "SecretKey" && isApiInstance === true)
         return tabbing + "var developerSecretKey = request.AuthenticationContext?.DeveloperSecretKey ?? (authenticationContext?.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey);\n            if (developerSecretKey == null) throw new PlayFabException(PlayFabExceptionCode.DeveloperKeyNotSet, \"DeveloperSecretKey is not found in Request, Server Instance or PlayFabSettings\");\n";
     if (apiCall.url === "/Authentication/GetEntityToken")
-        return tabbing + "string authKey = null, authValue = null;\n"
-            + tabbing + "if (PlayFabSettings.ClientSessionTicket != null) { authKey = \"X-Authorization\"; authValue = PlayFabSettings.ClientSessionTicket; }\n"
-            + tabbing + "if (PlayFabSettings.DeveloperSecretKey != null) { authKey = \"X-SecretKey\"; authValue = PlayFabSettings.DeveloperSecretKey; }\n"
-            + tabbing + "if (PlayFabSettings.EntityToken != null) { authKey = \"X-EntityToken\"; authValue = PlayFabSettings.EntityToken; }\n";
+        return tabbing + "string authKey = null, authValue = null;\n" 
+            + tabbing + "#if !DISABLE_PLAYFABCLIENT_API\n"
+            + tabbing + "        var clientSessionTicket = request.AuthenticationContext ?.ClientSessionTicket ?? PlayFabSettings.ClientSessionTicket;\n"
```

Comment:
I think this section will result in a "double tabbed" effect on these lines.
basically you have:
```
+ tabbing + "no spaces"
+ tabbing + "        eight spaces" 
```
in all of this section, and I expect that will produce improperly tabbed code in the destination.

Also please don't put a space when using "?.":
AuthenticationContext ?.ClientSessionTicket

In your C# repo, please open in Visual Studio, and press "Ctrl + K + D", to reformat the file in the VS Standard file spacing, and then make the template produce exactly that spacing.